src/Symfony/Component/HttpClient/HttpClientTrait.php

```diff
@@ -86,6 +86,14 @@ private static function prepareRequest(?string $method, ?string $url, array $opt
 
         if (isset($options['body'])) {
             $options['body'] = self::normalizeBody($options['body']);
+
+            if (\is_string($options['body'])
+                && (string) \strlen($options['body']) !== substr($h = $options['normalized_headers']['content-length'][0] ?? '', 16)
+                && ('' !== $h || ('' !== $options['body'] && !isset($options['normalized_headers']['transfer-encoding'])))
+            ) {
+                $options['normalized_headers']['content-length'] = [substr_replace($h ?: 'Content-Length: ', \strlen($options['body']), 16)];
```

Comment:
Can't we do `'Content-Length: '.\strlen($options['body'])` directly?

Reply:
We always try to preserve the case of the submitted headers, that's why.